core/tests/agnostic_tests.rs

```diff
@@ -1,35 +1,34 @@
+use anyhow::anyhow;
 use std::io::Write;
 use typeshare_core::{
     context::{ParseContext, ParseFileContext},
     language::{CrateTypes, Language, TypeScript},
-    parser::{self, ParseError},
-    rust_types::RustTypeParseError,
-    ProcessInputError,
+    parser::{self},
 };
+
 /// Parse and generate types for a single Rust input file.
 pub fn process_input(
     input: &str,
     language: &mut dyn Language,
     imports: &CrateTypes,
     out: &mut dyn Write,
-) -> Result<(), ProcessInputError> {
+) -> anyhow::Result<()> {
```

Comment:
For my own understanding (similar to the question about using `String` above), what benefits does this have over the existing approach of returning a typed error?